Unhandled merge options

**Describe the bug**
Merge mode options are not passed down to merge functions.

The command I executed:
`$> PYTHONPATH=gcovr-src python3 -m gcovr --json -o /build/folder/path/coverage/coverage.json --gcov-executable=/opt/x86_64-gcc_13.2.0-linux-gnu/bin/x86_64-gcc_13.2.0-linux-gnu-gcov -j -s --root '/path/to/source/' --exclude '/.*/(test|generated|build)/,*' --merge-mode-functions=merge-use-line-0 -- /build/folder/path`

It fails with the following stack trace:
```
(ERROR) Uncaught EXCEPTION
Traceback (most recent call last):
  File "/usr/lib64/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib64/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/build/folder/path/gcovr-src/gcovr/__main__.py", line 546, in <module>
    main()
  File "/build/folder/path/gcovr-src/gcovr/__main__.py", line 328, in main
    covdata = collect_coverage_from_gcov(options)
  File "/build/folder/path/gcovr-src/gcovr/__main__.py", line 392, in collect_coverage_from_gcov
    covdata = merge_covdata(covdata, context["covdata"])
  File "/build/folder/path/gcovr-src/gcovr/merging.py", line 190, in merge_covdata
    return _merge_dict(left, right, merge_file, options)
  File "/build/folder/path/gcovr-src/gcovr/merging.py", line 139, in _merge_dict
    _insert_coverage_item(left, key, right_item, merge_item, options)
  File "/build/folder/path/gcovr-src/gcovr/merging.py", line 175, in _insert_coverage_item
    merged_item = merge_item(target_dict[key], new_item, options)
  File "/build/folder/path/gcovr-src/gcovr/merging.py", line 218, in merge_file
    left.functions = _merge_dict(
  File "/build/folder/path/gcovr-src/gcovr/merging.py", line 139, in _merge_dict
    _insert_coverage_item(left, key, right_item, merge_item, options)
  File "/build/folder/path/gcovr-src/gcovr/merging.py", line 175, in _insert_coverage_item
    merged_item = merge_item(target_dict[key], new_item, options)
  File "/build/folder/path/gcovr-src/gcovr/merging.py", line 292, in merge_function
    raise AssertionError(
AssertionError: Got function <Code snippet removed> on multiple lines: 0, 35.
	You can run gcovr with --merge-mode-functions=MERGE_MODE.
	The available values for MERGE_MODE are described in the documentation.
```

If I change the call to `merge_covdata` on `__main__.py:392` to:
```
covdata = merge_covdata(covdata, context["covdata"], get_merge_mode_from_options(options))
```
it works as expected. Is this a bug, or are the merge options omitted for a reason?

**To Reproduce**
I have not been able to reproduce this issue on another codebase

**Expected behavior**
Since the error message suggests adding a command line option to handle the situation, the options should either be included or the error message should be changed


**Desktop (please complete the following information):**
 - OS: Rocky Linux 9
 - GCC version GCC 13.2
 - GCOVR version 6.0

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -49,7 +49,7 @@ Bug fixes and small improvements:
 - Fix reading of choices options from configuration files (e.g. ``gcov-ignore-parse-errors``). (:issue:`816`)
 - Fix ``TypeError`` during decision analysis. (:issue:`784`)
 - Use relative paths if possible when running ``gcov``. (:issue:`820`)
-- Respect :option`--merge-mode-functions`when merging coverage data. (:issue:`841`)
+- Respect :option`--merge-mode-functions`when merging coverage data. (:issue:`844`)
 
 Documentation:
 

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -49,6 +49,7 @@ Bug fixes and small improvements:
 - Fix reading of choices options from configuration files (e.g. ``gcov-ignore-parse-errors``). (:issue:`816`)
 - Fix ``TypeError`` during decision analysis. (:issue:`784`)
 - Use relative paths if possible when running ``gcov``. (:issue:`820`)
+- Use options when merging coverage data. (:issue:`841`)
 
 Documentation:
 

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -49,7 +49,7 @@ Bug fixes and small improvements:
 - Fix reading of choices options from configuration files (e.g. ``gcov-ignore-parse-errors``). (:issue:`816`)
 - Fix ``TypeError`` during decision analysis. (:issue:`784`)
 - Use relative paths if possible when running ``gcov``. (:issue:`820`)
-- Use options when merging coverage data. (:issue:`841`)
+- Respect :option`--merge-mode-functions`when merging coverage data. (:issue:`841`)
 
 Documentation:
 

--- a/gcovr/formats/gcov/read.py
+++ b/gcovr/formats/gcov/read.py
@@ -88,7 +88,9 @@ def read_report(options: Options) -> CovData:
     toerase = set()
     covdata = dict()
     for context in contexts:
-        covdata = merge_covdata(covdata, context["covdata"])
+        covdata = merge_covdata(
+            covdata, context["covdata"], get_merge_mode_from_options(options)
+        )
         toerase.update(context["toerase"])
 
     for filepath in toerase:
@@ -305,7 +307,6 @@ def guess_source_file_name_heuristics(
     starting_dir: str,
     obj_dir: str,
 ) -> str:
-
     # 0. Try using the path to the gcov file
     fname = os.path.join(os.path.dirname(data_fname), source_from_gcov)
     if os.path.exists(fname):

--- a/gcovr/merging.py
+++ b/gcovr/merging.py
@@ -179,9 +179,7 @@ def _insert_coverage_item(
     return merged_item
 
 
-def merge_covdata(
-    left: CovData, right: CovData, options: MergeOptions = DEFAULT_MERGE_OPTIONS
-) -> CovData:
+def merge_covdata(left: CovData, right: CovData, options: MergeOptions) -> CovData:
     """
     Merge CovData information.
 
@@ -202,7 +200,7 @@ def insert_file_coverage(
 def merge_file(
     left: FileCoverage,
     right: FileCoverage,
-    options: MergeOptions = DEFAULT_MERGE_OPTIONS,
+    options: MergeOptions,
 ) -> FileCoverage:
     """
     Merge FileCoverage information.
@@ -234,7 +232,7 @@ def insert_line_coverage(
 def merge_line(
     left: LineCoverage,
     right: LineCoverage,
-    options: MergeOptions = DEFAULT_MERGE_OPTIONS,
+    options: MergeOptions,
 ) -> LineCoverage:
     """
     Merge LineCoverage information.
@@ -273,7 +271,7 @@ def insert_function_coverage(
 def merge_function(
     left: FunctionCoverage,
     right: FunctionCoverage,
-    options: MergeOptions = DEFAULT_MERGE_OPTIONS,
+    options: MergeOptions,
 ) -> FunctionCoverage:
     """
     Merge FunctionCoverage information.
@@ -364,7 +362,7 @@ def insert_branch_coverage(
 def merge_branch(
     left: BranchCoverage,
     right: BranchCoverage,
-    options: MergeOptions = DEFAULT_MERGE_OPTIONS,
+    options: MergeOptions,
 ) -> BranchCoverage:
     """
     Merge BranchCoverage information.
@@ -391,7 +389,7 @@ def insert_call_coverage(
 def merge_call(
     left: CallCoverage,
     right: CallCoverage,
-    options: MergeOptions = DEFAULT_MERGE_OPTIONS,
+    options: MergeOptions,
 ) -> BranchCoverage:
     """
     Merge CallCoverage information.
@@ -409,14 +407,14 @@ def insert_decision_coverage(
     options: MergeOptions = DEFAULT_MERGE_OPTIONS,
 ) -> Optional[DecisionCoverage]:
     """Insert DecisionCoverage into LineCoverage."""
-    target.decision = merge_decision(target.decision, decision)
+    target.decision = merge_decision(target.decision, decision, options)
     return target.decision
 
 
 def merge_decision(
     left: Optional[DecisionCoverage],
     right: Optional[DecisionCoverage],
-    options: MergeOptions = DEFAULT_MERGE_OPTIONS,
+    options: MergeOptions,
 ) -> Optional[DecisionCoverage]:
     """
     Merge DecisionCoverage information.

